### DISCUSSION TITLE
Get all Userdata Class instances available

### DISCUSSION THREAD

**Discussion by yatoiv** [mlua-rs/mlua#480] [Q&A]
Let’s say I have a class called Person and it implements UserData trait, How can I track all available Persons assuming that lua code is still running and creates lots of Persons and they are strong references which are still in memory and not garbage collected

**Comment by cheesycod** [comment]
> Let’s say I have a class called Person and it implements UserData trait, How can I track all available Persons assuming that lua code is still running and creates lots of Persons and they are strong references which are still in memory and not garbage collected

Your best bet is to store a set of ids in a static variable (you can use dashmap/RwLock in a std::sync::LazyLock to achieve this). You can use a Drop trait impl to remove people from this static variable
